feat: support `@apollo/client` version 4

vince91 opened the pull request:
BREAKING CHANGE: drop support for `@apollo/client` version 3
BREAKING CHANGE: require `rxjs` version 7

Fixes https://github.com/DiederikvandenB/apollo-link-sentry/issues/489

vince91 committed:
chore: update apollo client and related

vince91 committed:
chore: fix jest config for new apollo client version

vince91 committed:
chore: update codebase for apollo client 4

spawnia commented on jest.config.js:
```diff
@@ -21,6 +21,9 @@ module.exports = {
 
   // A map from regular expressions to paths to transformers
   transform: {
+    '.(js|jsx)': 'babel-jest',
```
I think there is an `allowJs` option for `ts-jest` which might make this easier.

vince91 replied:
I have removed `transform` using `ts-jest` preset `ts-jest/presets/js-with-babel`.

Is it okay like that? I tried using `ts-jest/presets/js-with-ts` preset with `allowJs` in `tsconfig.json` as described in https://huafu.github.io/ts-jest/user/config/ but I was not able to run the tests...

spawnia reviewed (commented):
Looks pretty good overall! I am awaiting feedback on https://github.com/DiederikvandenB/apollo-link-sentry/issues/493 before I can consider merging this. In the meantime, you can address the minor feedback on jest setup I gave, and update the merge request title and description to support a proper semantic release (feat prefix, BREAKING CHANGE marker).

vince91 committed:
chore: remove transform in jest.config.js

vince91 commented:
> Looks pretty good overall! I am awaiting feedback on #493 before I can consider merging this. In the meantime, you can address the minor feedback on jest setup I gave, and update the merge request title and description to support a proper semantic release (feat prefix, BREAKING CHANGE marker).

Thanks for the review! I have updated the title and description and I have addressed your comment.

spawnia reviewed (approved):
I have updated the title and description to follow https://www.conventionalcommits.org/en/v1.0.0. This is good to go once our release process is fixed.

hunhejj commented:
@spawnia AFAICS this PR isn't be blocked anymore.

Alberto-BaseNet commented:
Any updates @spawnia ? trying to update apollo client to V4.

spawnia committed:
chore: resolve merge conflicts with master in yarn.lock
